### ISSUE TITLE
devon ide update does not pull latest project settings

### ISSUE THREAD

**Issue by andreasrichtercg** [devonfw/ide#363]
### Steps to reproduce (bug) / Use Case of feature request (enhancement)
Expected:
1. Architect creates project settings repository
2. Developer uses "setup" command  with settings repository url to setup devon ide with project settings
3. Architect updates project settings repository
4. Developer updates devon ide using "devon ide update"
5. Devon ide pulls latest settings from project settings
6. Devon ide updates components according latest settings defined by architect

Actual: 
1. Architect creates project settings repository
2. Developer uses "setup" command  with settings repository url to setup devon ide with project settings
3. Architect updates project settings repository
4. Developer updates devon ide using "devon ide update"
5. Devon ide does not pull latest settings from project settings
6. Devon ide updates components according initial settings state set defined by architect

### Affected version:
devonfw-ide-scripts 3.2.4
devonfw 3.2.1

* OS: Windows

**Comment by markusschuh** [comment]
I'd say, you indeed have identified a bug or important enhancement identified here. The `project settings` feature does not seem yet to get the awareness, it deserves. So any more use and opening of those bugs, is a very helpful feedback, IMHO.

The help of devon promises such an update - at least, when an explicit `update settings` is used:

```
devon help ide
. . . 
update settings                  update settings (git pull)
```

In order to analyze this, I have added now some quick-hack tracing functionality to my local  `devonfw-ide` scripts (see my comments in #261) which results in the output lines:

```
export DEVONFW_IDE_TRACE=1
devon ide update  2>&1 | less -N
. . .
   2288 + '[' update = -h ']'
   2289 + '[' update = help ']'
   2290 + '[' update = setup ']'
   2291 + '[' update = uninstall ']'
   2292 + '[' update = update ']'
   2293 + '[' -z '' ']'
   2294 + doUpdateSettings
   2295 + '[' '!' -d /e/opt/devonfw-ide-scripts/settings ']'
   2296 + doUpdateSoftware
. . .
```

When the `settings` directory exists, the `doUpdateSettings`  function should check, if it is a git repository and run a `git pull` then. But the related code - to be executed in the else clause of the "no settings directory exist?" check in above output line 2295 - does not yet seem to have been introduced into the [master](https://github.com/devonfw/ide/blob/09f98857db83d063a59e2537246920c7c39c8f19/scripts/src/main/resources/scripts/command/ide#L27) branch.

I assume the [doGitPullOrClone](https://github.com/devonfw/ide/blob/09f98857db83d063a59e2537246920c7c39c8f19/scripts/src/main/resources/scripts/functions#L720)  function in `functions` already provides most of the needed functionality, so a PR should not be too hard. But no PR from my side for today.

**Comment by hohwille** [comment]
Fixed with PR #367
